rwengine/src/audio/SoundBufferStreamed.hpp

```diff
@@ -0,0 +1,29 @@
+#ifndef _RWENGINE_SOUND_BUFFER_STREAMED_HPP_
+#define _RWENGINE_SOUND_BUFFER_STREAMED_HPP_
+
+#include "audio/SoundBuffer.hpp"
+
+#include <array>
+#include <future>
+
+struct SoundBufferStreamed : public SoundBuffer {
+    static constexpr unsigned int kNrBuffersStreaming = 4;
+    static constexpr unsigned int kSizeOfChunk = 4096;
+
+    SoundBufferStreamed();
+    ~SoundBufferStreamed() override;
+    bool bufferData(SoundSource& soundSource) final;
+
+    void play() final;
+    void pause() final;
+    void stop() final;
+
+private:
+    SoundSource* soundSource = nullptr;
+    void updateBuffers();
+    unsigned int streamedData = 0;
+    std::array<ALuint, kNrBuffersStreaming> buffers;
+    std::future<void> loadingThread;
```

Comment:
Nit naming wise: `SoundSource` is doing the loading, the async code here just uploads the data into OpenAL right?

Reply:
It feeds buffers and queue them (to be used).

Good point. :)